qiskit/transpiler/target.py

```diff
@@ -507,14 +530,17 @@ def operations_for_qargs(self, qargs):
                 instructions that apply to qubit 0.
         Returns:
             list: The list of :class:`~qiskit.circuit.Instruction` instances
-            that apply to the specified qarg.
+            that apply to the specified qarg. This may also be a class if
+            a variable width operation is globally defined.
 
         Raises:
             KeyError: If qargs is not in target
         """
         if qargs not in self._qarg_gate_map:
             raise KeyError(f"{qargs} not in target.")
-        return [self._gate_name_map[x] for x in self._qarg_gate_map[qargs]]
+        res = [self._gate_name_map[x] for x in self._qarg_gate_map[qargs]]
+        res += [self._gate_name_map[x] for x in self._qarg_gate_map[None]]
+        return res
```

Comment:
The `self._qarg_gate_map[None]` call here has the side-effect of adding `None` to the `defaultdict`, so `None in self._qarg_gate_map` will be `True` after this call, even if `None` wasn't originally there.  That _probably_ doesn't cause any problems, but it might be safer to avoid it.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/8852/commits/e1b72bd35cd47cfdb55b562eca2e98e277240a1b